warehouse/warehousegrpc.go

```diff
@@ -94,12 +97,49 @@ func (*warehouseGRPC) TriggerWHUpload(_ context.Context, request *proto.WHUpload
 	return res, err
 }
 
-func (grpc *warehouseGRPC) Validate(_ context.Context, req *proto.WHValidationRequest) (*proto.WHValidationResponse, error) {
-	handleT := validations.CTHandleT{
```

Comment:
The domain model for the validations package should depend on the destination only, not on the internal representation of the destination like whether tunneling is enabled or not. 
Moving the logic of adding those information to the caller itself